qubes/vm/qubesvm.py

```diff
@@ -1445,28 +1448,30 @@ async def run_service(self, service, source=None, user=None,
 
         source = 'dom0' if source is None else self.app.domains[source].name
 
+        name = self.name + '-dm' if stubdom else self.name
+
         if user is None:
             user = self.default_user
 
         if self.is_paused():
             # XXX what about autostart?
             raise qubes.exc.QubesVMNotRunningError(
-                self, 'Domain {!r} is paused'.format(self.name))
+                self, 'Domain {!r} is paused'.format(name))
         if not self.is_running():
             if not autostart:
                 raise qubes.exc.QubesVMNotRunningError(self)
             await self.start(start_guid=gui)
 
         if not self.is_qrexec_running():
```

Comment:
This is a bit unfortunate - it checks if qrexec is connected to the actual domain, not to the stubdomain, even if the call is requested to the stubdomain (`stubdom=True`).